skbio/stats/distance/tests/test_permdisp.py

```diff
@@ -0,0 +1,253 @@
+        self.assertAlmostEqual(exp_stat, obs_mixed, places=6)
+
+    def test_centroids_null(self):
+        dm = pcoa(self.null_mat)
+        dm = dm.samples
+
+        obs_null = _compute_centroid_groups(dm, self.grouping_eq)
+        np.isnan(obs_null)
+
+    def test_centroid_normal(self):
+        exp = pd.Series(index=self.exp_index,
+                        data=['PERMDISP', 'F-value', 9, 2, 0.244501519876,
+                              0.63, 99],
+                        name='PERMDISP results')
+
+        grouping = ['Control', 'Control', 'Control', 'Control', 'Control',
+                    'Fast', 'Fast', 'Fast', 'Fast']
+
+        np.random.seed(0)
+        obs = permdisp(self.unifrac_dm, grouping, test='centroid',
+                       permutations=99)
+
+        self.assert_series_equal(obs, exp)
+
+    def test_median_normal(self):
+
+        exp = pd.Series(index=self.exp_index,
+                        data=['PERMDISP', 'F-value', 9, 2, 0.139475441876,
+                              0.61, 99],
+                        name='PERMDISP results')
+
+        np.random.seed(0)
+        obs = permdisp(self.unifrac_dm, self.unif_grouping, test='median',
+                       permutations=99)
+
+        self.assert_series_equal(obs, exp)
+
+    def test_not_distance_matrix(self):
+        dm = []
+        grouping = ['Control', 'Control', 'Control', 'Control', 'Control',
+                    'Fast', 'Fast', 'Fast', 'Fast']
+
+        npt.assert_raises(TypeError, permdisp, dm, grouping, permutations=0)
+
+    def test_mismatched_group(self):
+
+        gr = ['foo', 'bar']
+        npt.assert_raises(ValueError, permdisp, self.unifrac_dm, gr)
+
+    def test_single_group(self):
+
+        gr = ['f', 'f', 'f', 'f', 'f', 'f', 'f', 'f', 'f']
+        npt.assert_raises(ValueError, permdisp, self.unifrac_dm, gr)
+
+    def test_no_permuations(self):
+        obs = permdisp(self.eq_mat, self.grouping_eq, permutations=0)
+
+        # fix this to test both are significant or not significant
+        pval = obs['p-value']
+        np.isnan(pval)
+
+    def test_hdmedians(self):
+        exp = np.array([2.01956244, 1.53164546, 2.60571752, 0.91424179,
+                        1.76214416, 1.69943057])
+        obs = np.array(hd.geomedian(self.eq_mat.data))
+        npt.assert_almost_equal(obs, exp, decimal=6)
+
+    def test_confirm_betadispr_results(self):
+        mp_dm = DistanceMatrix.read('data/moving_pictures_dm.tsv')
+        mp_mf = pd.read_csv('data/moving_pictures_mf.tsv', sep='\t')
+        mp_mf.set_index('#SampleID', inplace=True)
+
+        exp_med_mp = 10.4105
+        exp_cen_mp = 17.727
+        obs_med_mp = permdisp(mp_dm, mp_mf,
+                              column='BodySite')['test statistic']
+        obs_cen_mp = permdisp(mp_dm, mp_mf, column='BodySite',
+                              test='centroid')['test statistic']
+
+        self.assertAlmostEqual(exp_med_mp, obs_med_mp, places=0)
```

Comment:
Can you test for the whole pd.Series object? While you do that can you validate that the p-value doesn't change in significance i.e. that both are <0.05 or >0.05.

Reply:
great idea!